activerecord/lib/active_record/type/date.rb

```diff
@@ -9,6 +9,10 @@ def klass
         ::Date
       end
 
+      def type_cast_for_database(value)
```

Comment:
> I suspect that because the `where` clause uses an SQL string, it may not be able to determine the column type?

Right, AR cannot determine the column type if using `where` with an SQL string for now.